scene/gui/label.cpp

```diff
@@ -879,11 +883,17 @@ String Label::get_text() const {
 void Label::set_visible_characters(int p_amount) {
 	if (visible_chars != p_amount) {
 		visible_chars = p_amount;
-		if (get_total_character_count() > 0) {
-			visible_ratio = (float)p_amount / (float)get_total_character_count();
-		} else {
+		if (p_amount < 0) { // special case when setting or resetting value to negative (unlimited) via the UI, fixes https://github.com/godotengine/godot/issues/83387
 			visible_ratio = 1.0;
+		} else {
+			// calculate the ratio for the user, if defined. This works even if the text is empty because 1.0 is the default, and defaults aren't seialized.
```

Comment:
```suggestion
			// Calculate the ratio for the user, if defined. This works even if the text is empty because 1.0 is the default, and defaults aren't serialized.
```